tests/host/common/MockUDPSocket.cpp

```diff
@@ -0,0 +1,182 @@
+/*
```

Comment:
I didn't mean the filename, although that's ok. I was referring to the function names. Instead of having mockSomething() mockOtherthing(), you could do 
```
namespace mock
{
void something();
void otherthing();
}
```
and eventually, you could do the following for equivalent functionality with different implementation on the host:
```
namespace host
{
class ClientContext
{...};
}
```
which could be the counterpart to `namespace esp8266 {class ClientContext {...}; }`.
Then, you can map into the global namespace as needed.

Reply:
I think @d-a-v is going to run into many *.C files, where `namespace` isn't really going to be usable.  So maybe mockXXX is good enough, and will be consistent?

Reply:
Ok, agreed.